plugins/utils/which-key.nix

```diff
@@ -55,257 +265,308 @@ with lib;
+    };
+
+    show_help = defaultNullOpts.mkBool true "show a help message in the command line for using WhichKey";
+
+    show_keys = defaultNullOpts.mkBool true "show the currently pressed key and its label as a message in the command line";
 
     disable = {
-      buftypes = helpers.defaultNullOpts.mkListOf types.str [ ] "Disabled by default for Telescope";
-      filetypes = helpers.defaultNullOpts.mkListOf types.str [ ] "";
+      bt = defaultNullOpts.mkListOf types.str [ ] "Disabled by default for Telescope";
+      ft = defaultNullOpts.mkListOf types.str [ ] "";
     };
+
+    debug = defaultNullOpts.mkBool false "Enable wk.log in the current directory.";
   };
 
-  config =
-    let
-      cfg = config.plugins.which-key;
-      opt = options.plugins.which-key;
-      setupOptions = {
-        plugins = {
-          inherit (cfg.plugins) marks registers spelling;
-          presets = {
-            inherit (cfg.plugins.presets) operators motions;
-            text_objects = cfg.plugins.presets.textObjects;
-            inherit (cfg.plugins.presets)
-              windows
-              nav
-              z
-              g
-              ;
-          };
-        };
-        inherit (cfg) operators;
-        key_labels = cfg.keyLabels;
-        inherit (cfg) motions icons;
-        popup_mappings = {
-          scroll_down = cfg.popupMappings.scrollDown;
-          scroll_up = cfg.popupMappings.scrollUp;
-        };
-        window =
-          let
-            mkSpacing =
-              opt:
-              helpers.ifNonNull' opt [
-                opt.top
-                opt.right
-                opt.bottom
-                opt.top
-              ];
-          in
-          {
-            inherit (cfg.window) border position;
-            margin = mkSpacing cfg.window.margin;
-            padding = mkSpacing cfg.window.padding;
-            inherit (cfg.window) winblend;
-          };
-        inherit (cfg) layout;
-        ignore_missing = cfg.ignoreMissing;
-        inherit (cfg) hidden;
-        show_help = cfg.showHelp;
-        show_keys = cfg.showKeys;
-        inherit (cfg.settings) spec;
-        inherit (cfg) triggers;
-        triggers_nowait = cfg.triggersNoWait;
-        triggers_blacklist = cfg.triggersBlackList;
-        inherit (cfg) disable;
-      };
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
+  settingsExample = {
+    preset = false;
 
-      warnings = optional opt.registrations.isDefined ''
-        nixvim (plugins.which-key):
-        The option definition `plugins.which-key.registrations' in ${showFiles opt.registrations.files} has been deprecated in which-key v3; please remove it.
-        You should use `plugins.which-key.settings.spec' instead.
+    delay = 200;
 
-        Note: the spec format has changed in which-key v3
-        See: https://github.com/folke/which-key.nvim?tab=readme-ov-file#%EF%B8%8F-mappings
-      '';
+    spec = [
+      {
+        __unkeyed = "<leader>b";
```

Comment:
While any key starting with `__unkeyed` will work, can we use `__unkeyed-1` in examples since that matches what `lib.nixvim.listToUnkeyedAttrs` will spit out.